### PR TITLE
✨ Add AWS VPC subnet resource

### PR DESCRIPTION
**Pull request by mariuskimmina**
#1794

first time making changes to providers, so please review carefully. 
Example usage

image

I wasn't sure what kind of fields to add here. Just went with `id` and `cidrs` for now - there's probably more that could make sense.

### CODE REVIEW

**Comment by mariuskimmina** [comment]
Added the filed `mapPublicIpOnLaunch` which is the actual requirement for AWS security recommendations discussed in #1794

```
cnquery> aws.vpcs.where( arn == "arn:aws:vpc:eu-central-1:455954587013:id/vpc-091edd43468ca60e5" )[0].subnets[0].mapPublicIpOnLaunch
[ok] value: true
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 1]
```diff
@@ -1423,6 +1423,7 @@ resources:
       region: {}
       routeTables: {}
       state: {}
+      subnets: {}
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
      subnets:
        min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 2]
```diff
@@ -1450,3 +1451,13 @@ resources:
     platform:
       name:
       - aws
+  aws.vpc.subnet:
+    fields:
+      cidrs: {}
+      id: {}
+      mapPublicIpOnLaunch: {}
+    is_private: true
+    min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

**Review by vjeffrey — CHANGES_REQUESTED** [review]
can you please add in the subnet arn, and add an id function for the subnet `func (s *mqlAwsVpcSubnet) id() (string, error) {` that returns the arn?

this will help ensure the objects are uniquely identifiable

**Comment by mariuskimmina** [comment]
I have added the arn filed, I noticed a bug tho.
Currently I have 3 subnets in AWS, the output in cnquery looks like this

aws.vpcs.where[0].subnets: [
  0: aws.vpc.subnet id="subnet-021eb62bc781a5ed5"
  1: aws.vpc.subnet id="subnet-021eb62bc781a5ed5"
  2: aws.vpc.subnet id="subnet-021eb62bc781a5ed5"
]

So the number of subnets is correct but they all have the same id.
I copied this part from the exisiting routeTables function and found that the same bug exists there as well

I have 2 route tables in my aws account

cnquery> aws.vpcs.where( arn == "arn:aws:vpc:eu-central-1:455954587013:id/vpc-091edd43468ca60e5" )[0].routeTables
aws.vpcs.where[0].routeTables: [
  0: aws.vpc.routetable id="rtb-0d28b6f55dfe767c2"
  1: aws.vpc.routetable id="rtb-0d28b6f55dfe767c2"
]

I see two route tables here but they both have the same id

**Comment by vjeffrey** [comment]
did you add in the id function?
```
func (s *mqlAwsVpcSubnet) id() (string, error) {
  return s.Arn.Data, nil
}
```

route table might also need the id fix

**Comment by mariuskimmina** [comment]
> did you add in the id function?
> 
> ```
> func (s *mqlAwsVpcSubnet) id() (string, error) {
>   return s.Arn.Data, nil
> }
> ```
> 
> route table might also need the id fix

Ah, adding the `id` function does seem to fix it. 

I also saw now this comment
```
to override the _id implement id()
```

But, doesn't that mean that there is probably a bug in the default _id implementation because it should have been able to use the correct `id` field for each subnet and not the same one three times?

**Comment by vjeffrey** [comment]
using the arn as the id guarantees we get the unique one since it includes the region and account id - whenever dealing with aws we always want to use the arn where possible because of that.

good question about the _id thing tho, i haven't looked at it much yet in our new v9 world...i'll dig around for an answer for u on that

**Comment by mariuskimmina** [comment]
Anything still needs to be done here from my side? @vjeffrey

**Comment by vjeffrey** [comment]
sorry for the delay! this is great, thank you!

**Review by vjeffrey — APPROVED** [review]

**Pull request merged**